client/app/controllers/collection/add.js

```diff
@@ -1,19 +1,127 @@
 import Ember from 'ember';
 
+function updateState() {
+    console.log('updating state');
+    const state = this.get('state')
+    this.get('actions').forEach((action) => {
+
+        function condition_dispatcher(condition) {
+
+            // Check if its a regular condition
+            if (condition.parameter !== undefined) {
+                // actualy check the condition is met;
+                // the parameter has to have the given state.
+                if (state[condition.parameter] === undefined) {
+                    state[condition.parameter] = {};
+                }
+                if (state[condition.parameter].state === undefined) {
+                    state[condition.parameter].state = [];
+                }
+                const parameter_state = state[condition.parameter].state
+                // check that the state exists for this item
+                return parameter_state.some((state_item) => state_item === condition.state)
+            }
+
+            // Check if its an 'all' composite condition
+            if (condition.all !== undefined &&
+                condition.all.constructor === Array
+            ) {
+                // if any conditions fail, the whole check fails.
+                return check_all(condition.all);
+            }
+
+            // Check if its an 'any' composite condition
+            if (condition.any !== undefined &&
+                condition.any.constructor === Array
+            ) {
+                // if any conditions are met, the whole check passes.
+                return check_any(condition.any);
+            }
+
+            // Check if its a 'none' composite condition
+            if (condition.none !== undefined &&
+                condition.none.constructor ===Array
+            ) {
+                // if any conditions are met, the whole check fails.
+                return !check_any(condition.none);
+            }
+
+            return false;
+
+        }
+
+        function check_all(conditions) {
+            // if any conditions fail, the whole check fails.
+            return !conditions.some(function(condition) {
+                return !condition_dispatcher(condition);
+            });
+        }
+
+        function check_any(conditions) {
+            // if any conditions are met, the whole check passes.
+            return action.conditions.some(condition_dispatcher)
```

Comment:
This function should use the `conditions` parameter instead of `action.conditions`. Also, what is a use case in which only some of the conditions need to be met?